server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilderTests.java

```diff
@@ -119,10 +121,18 @@ public void testSupportsParallelCollection() {
             new CompositeAggregationBuilder(randomAlphaOfLength(10), Collections.singletonList(randomHistogramSourceBuilder()))
                 .supportsParallelCollection(null)
         );
-        assertTrue(
-            new CompositeAggregationBuilder(randomAlphaOfLength(10), Collections.singletonList(randomGeoTileGridValuesSourceBuilder()))
-                .supportsParallelCollection(null)
+        CompositeAggregationBuilder builder = new CompositeAggregationBuilder(
+            randomAlphaOfLength(10),
+            Collections.singletonList(randomGeoTileGridValuesSourceBuilder())
         );
+        assertTrue(builder.supportsParallelCollection(null));
+        builder.subAggregation(new TermsAggregationBuilder("name") {
+            @Override
+            public boolean supportsParallelCollection(ToLongFunction<String> fieldCardinalityResolver) {
+                return false;
+            }
+        });
+        assertFalse(builder.supportsParallelCollection(null));
```

Comment:
Here I added a missing test that verifies that we still look at sub-aggs for composite agg, when all the sources support concurrency.